From 59c90eceb8b912d96f6f38e9c1127ff464c9166b Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:07 -0400
Subject: [tools] lib/traceevent: Get rid of malloc_or_die() in pevent_filter_alloc()

Message-id: <1408953210-26343-120-git-send-email-jolsa@redhat.com>
Patchwork-id: 88822
O-Subject: [PATCH RHEL7.1 BZ1133083 119/322] tools lib traceevent: Get rid of malloc_or_die() in pevent_filter_alloc()
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 4f24416331e9a507e953e90d4534e9a9802cbc12
Author: Namhyung Kim <namhyung@kernel.org>
Date: Mon Dec 9 14:34:00 2013 +0900

description
===========
It returns NULL when allocation fails so the users should check the
return value from now on.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/parse-filter.c b/tools/lib/traceevent/parse-filter.c
index 2500e75..b3a61d4 100644
--- a/tools/lib/traceevent/parse-filter.c
+++ b/tools/lib/traceevent/parse-filter.c
@@ -182,7 +182,10 @@ struct event_filter *pevent_filter_alloc(struct pevent *pevent)
 {
  struct event_filter *filter;
 
- filter = malloc_or_die(sizeof(*filter));
+ filter = malloc(sizeof(*filter));
+ if (filter == NULL)
+  return NULL;
+
  memset(filter, 0, sizeof(*filter));
  filter->pevent = pevent;
  pevent_ref(pevent);
-- 
1.7.1